fe/fe-core/src/main/java/org/apache/doris/catalog/Tablet.java

```diff
@@ -227,10 +228,25 @@ public void getQueryableReplicas(List<Replica> allQuerableReplica, long visibleV
                 // replica.getSchemaHash() == -1 is for compatibility
                 if (replica.checkVersionCatchUp(visibleVersion, false)
                         && (replica.getSchemaHash() == -1 || replica.getSchemaHash() == schemaHash)) {
-                    allQuerableReplica.add(replica);
+                    allQueryableReplica.add(replica);
                 }
             }
         }
+
+        if (Config.skip_compaction_slower_replica && allQueryableReplica.size() > 1) {
+            long minVersionCount = Long.MAX_VALUE;
+            for (Replica replica : allQueryableReplica) {
+                if (replica.getVersionCount() != -1 && replica.getVersionCount() < minVersionCount) {
```

Comment:
check replica.getVersionCount() > 0 is better. Since the tablet has at least one rowset